### ISSUE TITLE
VSZ pidstat manpage has maybe wrong unit

### ISSUE THREAD

**Issue by gira0** [sysstat/sysstat#399]
Hello everyone, was recenty confused while comparing some tools that the man page of pidstat claims that VSZ is kilobytes but should this not be kibibytes. I'm not really great with the C language but shold this line not divide by 2^10, so 1024?
https://github.com/sysstat/sysstat/blob/6a976eb60aed3773076ffd99fd0058c74678ccde/pidstat.c#L491

Line in the man page:
https://github.com/sysstat/sysstat/blob/6a976eb60aed3773076ffd99fd0058c74678ccde/man/pidstat.1#L172

As an example, top and pidstat report the same value for VIRT/VSZ and top claims that it reports in "KiB" so something is going on here.

**Comment by neilser** [comment]
The [FAQ](https://github.com/sysstat/sysstat/blob/master/FAQ.md) comments on this:

> 1.4. What are the units actually used by sysstat commands?
> 
> A: Although sysstat commands use the following abbreviations: kB, MB, etc. as part of the metrics names (e.g. kB_read/s, wkB/s, or even sometimes with a lower 'b': kbmemfree, kbavail...), and the manual pages speak of kilobytes, megabytes, etc., we always actually refer to kibibytes (kiB), mebibytes (MiB),...
>A kibibyte is equal to 1024 bytes, and a mebibyte is equal to 1024 kibibytes.
>Metrics names have been defined many years ago. We don't modify them to avoid breaking third-party programs parsing sysstat commands' output.

You may not be aware that "kilobyte" was in widespread use for many years (and meant 1024 bytes) before the IEC invented "kibibyte" (etc.), and it was many more years after that before the new names achieved any real penetration into the tech world.

**Comment by gira0** [comment]
I was essentially born around the time the IEC made this a standard, so I’ve never experienced a world where kilobyte meant 1024 bytes. Are there any plans to update this? From my perspective, it’s a bit confusing, especially since other tools like top seem to have adopted kibibytes. I’ll admit I skipped over the FAQ, but relying solely on it to address this might not be the most user-friendly solution.

**Comment by neilser** [comment]
Yeah, I'd agree it's not ideal, but tbh I personally wish that they had never invented the binary names in the first place.

For a _very_ long time (I feel like it was _at least_ a decade beyond when the IEC apparently came up with the names, which was 1998 according to Wikipedia), there was really no confusion at all in techy circles about what kB, MB, GB, TB etc. meant, and (in my experience) no penetration of the new names either.

The sole exception to this unambiguity was when _buying_ (not so much _using_) a hard disk - those particular manufacturers had for many years chosen the cheesy option of decimal sizing because that made their products look better (it was such a minor win in the MB era that I really never understood that very bad call).
_Using_ a hard disk pretty much always involved using the binary names, which led to odd/broken things like "formatting" being blamed for a "1 TB" drive showing up as "930 GB" in one's OS of choice, lol.

Anyway, I digress. I think the FAQ makes it plain that this will not be addressed for sysstat, but to get a better answer you'll need to hear from the nice chap (Sebastien) who maintains the package.

**Comment by sysstat** [comment]
I couldn't have explained my position better than what did @neilser in his post above! Thank you guy :wink:
Yet, even if I don't modify the metrics names in the various sysstat's reports for the reason explained in the FAQ, I admit that I should probably now use the binary units in the man pages. Maybe a future patch to come...
